### PR TITLE
Implement listGroups

### LINKED ISSUE #18: Membership: List Groups
https://www.vinyldns.io/api/list-groups.html

### PR DESCRIPTION
**Pull request by mitruly**
Code changes:
- Add `Group` model
- Implement `listGroups`

Resolves #18.

### CODE REVIEW

File: src/main/java/io/vinyldns/java/model/membership/Group.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package io.vinyldns.java.model.membership;
+
+import java.util.Set;
+import org.joda.time.DateTime;
+
+public class Group {
+    private String name, email, description, id;
```
Comment (pauljamescleary) [inline src/main/java/io/vinyldns/java/model/membership/Group.java]: mark the non optional fields (fields that are set in all the constructors) as final

**Comment by mitruly** [comment]
Tested this against the API running on `localhost`:
```
    VinylDNSClientImpl vinylClient =
        new VinylDNSClientImpl(
            new VinylDNSClientConfig(
                "http://localhost:9000",
                new BasicAWSCredentials(<access_key>, <secret_key>)));

    VinylDNSResponse<ListGroupsResponse> vinylResponse = vinylClient.listGroups(new ListGroupsRequest());
    System.out.println("response: " + vinylResponse.toString());
```

Received the following response:
```
VinylDNSSuccessResponse{value=ListGroupsResponse{groups='[Group{name='test group', email='asda', description='adsa', id='6f679996-c904-43e9-8ea3-7e3789694fbf', status=Active, memberIds=null, adminUserIds=null}]', groupNameFilter='null', startFrom='null', nextId='null', maxItems=100}, statusCode=200}
```

File: src/main/java/io/vinyldns/java/model/membership/Group.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
+package io.vinyldns.java.model.membership;
+
+import java.util.Set;
+import org.joda.time.DateTime;
+
+public class Group {
+    private final String name, email, id;
+    private String description;
```
Comment (pauljamescleary) [inline src/main/java/io/vinyldns/java/model/membership/Group.java]: I would leave`description` as `final` and just explicitly set it to `null` in the other constrctor

**Comment by pauljamescleary** [comment]
One small comment, but you have the same git author issue here as on the other PR.  Gotta amend author for the CLA to be picked up

File: src/main/java/io/vinyldns/java/model/membership/Group.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        this.email = email;
+        this.description = description;
+        this.id = id;
+        this.created = created;
+        this.status = status;
+        this.memberIds = memberIds;
+        this.adminUserIds = adminUserIds;
+    }
+
+    public Group(String name, String email, String id, DateTime created, GroupStatus status,
```
Comment (pauljamescleary) [inline src/main/java/io/vinyldns/java/model/membership/Group.java]: not good form to take a parameter out of the middle of the constructor (seems like you removed description here), especially as it is backfilled with another `String`.

Leave "optional" elements to the end of the constructor, then set them to a valid default value in the constructor itself.

Often times it is good form to have one constructor call another providing the defaults, as opposed to the `this.name = name` bits we see here.

Look at the default/optional params in the API for guidance.

I am pretty sure that `id`, `created`, `status` should all be optional.  `memberIds` and `adminUserIds` can default to an empty set (even though they will fail API side).  We don't have a mechanism for throwing errors in this client so I am actually ok with that.

**Pull request merged**